### PR TITLE
std::vector to SmallVector

### PR DESCRIPTION
**Pull request by tgymnich**

### CODE REVIEW

File: enzyme/Enzyme/AdjointGenerator.h [hunk 1]
```diff
@@ -4252,8 +4252,8 @@ class AdjointGenerator
     SmallVector<Value *, 8> pre_args = {0, 0, 0};
     std::vector<DIFFE_TYPE> argsInverted = {DIFFE_TYPE::CONSTANT,
                                             DIFFE_TYPE::CONSTANT};
-    std::vector<Instruction *> postCreate;
-    std::vector<Instruction *> userReplace;
+    SmallVector<Instruction *> postCreate;
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:4255]: SmallVector requires a size in the type for lower llvm versions

File: enzyme/Enzyme/AdjointGenerator.h [hunk 2]
```diff
@@ -4252,8 +4252,8 @@ class AdjointGenerator
     SmallVector<Value *, 8> pre_args = {0, 0, 0};
     std::vector<DIFFE_TYPE> argsInverted = {DIFFE_TYPE::CONSTANT,
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:4253]: Change this vector too?

File: enzyme/Enzyme/AdjointGenerator.h (hunk 1 above)
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h]: fixed

File: enzyme/Enzyme/AdjointGenerator.h (hunk 2 above)
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h]: done

Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h:4259]: turns out it is really annoying to have a `SmallVector` instance inside the cache keys.

File: enzyme/Enzyme/AdjointGenerator.h [hunk 3]
```diff
@@ -11459,7 +11463,7 @@ class AdjointGenerator
             cast<Function>(called), subretType, argsInverted,
             TR.analyzer.interprocedural, /*returnValue*/ subretused, Mode,
             ((DiffeGradientUtils *)gutils)->FreeMemory, gutils->getWidth(),
-            tape ? tape->getType() : nullptr, nextTypeInfo, {},
+            tape ? tape->getType() : nullptr, nextTypeInfo, uncacheable_args,
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:11466]: Why did this not break before? is there an easy test

**Pull request merged**